### PR TITLE
Added test cases for Activator.CreateInstance<T>

### PR DESCRIPTION
**Pull request by YohDeadfall**
This is designed to cover most use cases of the generic `CreateInstance` method except references and pointers. It's prerequisite for merging dotnet/coreclr#24034.

 /cc @jkotas

### CODE REVIEW

File: src/System.Runtime/tests/System/ActivatorTests.Generic.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
+﻿using System;
```
Comment (jkotas) [inline src/System.Runtime/tests/System/ActivatorTests.Generic.cs]: Could you please add standard license header?

File: src/System.Runtime/tests/TestStructs/System.TestStructs.il [hunk 2]
```diff
@@ -0,0 +1,69 @@
+.assembly extern System.Runtime
```
Comment (jkotas) [inline src/System.Runtime/tests/TestStructs/System.TestStructs.il]: Same here

File: src/System.Runtime/tests/TestStructs/System.TestStructs.ilproj [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.IL">
+  <PropertyGroup>
+    <AssemblyVersion>1.0.0.0</AssemblyVersion>
+    <ProjectGuid>{6B002B34-089C-4BC4-91DD-57D350DEA91C}</ProjectGuid>
+    <Configurations>netstandard-Debug;netstandard-Release</Configurations>
+    <TestStrongNameKeyId>Microsoft</TestStrongNameKeyId>
```
Comment (jkotas) [inline src/System.Runtime/tests/TestStructs/System.TestStructs.ilproj]: Is the "TestStrongNameKeyId" actually needed?

**Comment by jkotas** [comment]
LGTM otherwise. Thank you!

File: src/System.Runtime/tests/TestStructs/System.TestStructs.ilproj (hunk 3 above)
Comment (YohDeadfall) [reply src/System.Runtime/tests/TestStructs/System.TestStructs.ilproj]: Yes, without it I'm getting `Error: ilasm on CoreCLR does not support full sign.`

**Comment by YohDeadfall** [comment]
Addressed review and rebased on the top of `master`.

**Pull request merged**